runtime/codert_vm/cnathelp.cpp

```diff
@@ -1745,6 +1745,37 @@ old_slow_jitResolveInvokeDynamic(J9VMThread *currentThread)
 	return addr;
 }
 
+void* J9FASTCALL
+old_slow_jitResolveConstantDynamic(J9VMThread *currentThread)
+{
+	OLD_SLOW_ONLY_JIT_HELPER_PROLOGUE(3);
+	DECLARE_JIT_PARM(J9ConstantPool*, ramConstantPool, 1);
+	DECLARE_JIT_INT_PARM(cpIndex, 2);
+	DECLARE_JIT_PARM(void*, jitEIP, 3);
+	void *addr = NULL;
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramConstantPool + cpIndex;
+	j9object_t *constantDynamicValue = &ramCPEntry->value;
+
+	/* Check if the value is resolved, Void.Class exception represents a valid null reference */
+	if (NULL == *constantDynamicValue && ramCPEntry->exception != currentThread->javaVM->voidReflectClass->classObject) {
+		/* If entry resolved to an exception previously, same exception will be set by resolution code */
+		buildJITResolveFrameWithPC(currentThread, J9_SSF_JIT_RESOLVE_DATA, parmCount, true, 0, jitEIP);
+		constantDynamicValue = &(currentThread->javaVM->internalVMFunctions->resolveConstantDynamic(
```

Comment:
This likely won't even compile, and certainly doesn't do what you want. Just refetch the pointer after the addr NULL check.